code/modules/industrial_lift/tram_controls.dm

```diff
@@ -172,6 +187,53 @@
 
 	computer.try_send_tram(destination)
 
+/obj/machinery/computer/tram_controls/proc/set_controls_display()
+	var/datum/lift_master/tram/tram = tram_ref?.resolve()
+
+	if(!tram || !is_operational)
+		return PROCESS_KILL
+
+	use_power(active_power_usage)
+
+	if(!tram.travelling)
+		if(istype(tram.from_where, /obj/effect/landmark/tram/left_part))
+			icon_screen = "[base_icon_state][DESTINATION_WEST_IDLE]"
+			previous_destination = tram.from_where
+			update_appearance()
+			return PROCESS_KILL
+
+		if(istype(tram.from_where, /obj/effect/landmark/tram/middle_part))
+			icon_screen = "[base_icon_state][DESTINATION_CENTRAL_IDLE]"
+			previous_destination = tram.from_where
+			update_appearance()
+			return PROCESS_KILL
+
+		if(istype(tram.from_where, /obj/effect/landmark/tram/right_part))
+			icon_screen = "[base_icon_state][DESTINATION_EAST_IDLE]"
+			previous_destination = tram.from_where
+			update_appearance()
+			return PROCESS_KILL
+
+	if(istype(tram.from_where, /obj/effect/landmark/tram/left_part))
+		previous_destination = tram.from_where
+		icon_screen = "[base_icon_state][DESTINATION_WEST_ACTIVE]"
+		update_appearance()
+		return PROCESS_KILL
+
+	if(istype(tram.from_where, /obj/effect/landmark/tram/middle_part))
+		if(istype(previous_destination, /obj/effect/landmark/tram/left_part))
+			icon_screen = "[base_icon_state][DESTINATION_CENTRAL_EASTBOUND_ACTIVE]"
+		if(istype(previous_destination, /obj/effect/landmark/tram/right_part))
+			icon_screen = "[base_icon_state][DESTINATION_CENTRAL_WESTBOUND_ACTIVE]"
+		update_appearance()
+		return PROCESS_KILL
+
+	if(istype(tram.from_where, /obj/effect/landmark/tram/right_part))
+		previous_destination = tram.from_where
+		icon_screen = "[base_icon_state][DESTINATION_EAST_ACTIVE]"
+		update_appearance()
+		return PROCESS_KILL
```

Comment:
this code smells wrong. Lots of lines doing the same thing in different places. it's a bad pattern, makes things hard to modify.
You could make the icon state extension to use a var and use else if, and reduce this down from 37 lines to I bet like 15

Reply:
Yeah, made it much more simple, now we're down to a single if statement.